java/src/main/java/org/hyperledger/fabric/client/GatewayImpl.java

```diff
@@ -148,14 +148,20 @@ public Network getNetwork(final String networkName) {
 
     @Override
     public Proposal newSignedProposal(final byte[] bytes, final byte[] signature) {
+        ProposalImpl result = (ProposalImpl) newProposal(bytes);
+        result.setSignature(signature);
+        return result;
+    }
+
+    @Override
+    public Proposal newProposal(final byte[] bytes) {
         try {
             ProposedTransaction proposedTransaction = ProposedTransaction.parseFrom(bytes);
             ProposalPackage.Proposal proposal = ProposalPackage.Proposal.parseFrom(proposedTransaction.getProposal().getProposalBytes());
             Common.Header header = Common.Header.parseFrom(proposal.getHeader());
             Common.ChannelHeader channelHeader = Common.ChannelHeader.parseFrom(header.getChannelHeader());
 
             ProposalImpl result = new ProposalImpl(client, signingIdentity, channelHeader.getChannelId(), proposedTransaction);
-            result.setSignature(signature);
             return result;
```

Comment:
If the `result` variable is only used as a return value in the following line, it is unnecessary. Just return the value directly:

```java
return new ProposalImpl(client, signingIdentity, channelHeader.getChannelId(), proposedTransaction);
```